src/ranch_server_proxy.erl

```diff
@@ -0,0 +1,48 @@
+-module(ranch_server_proxy).
+
+-behavior(gen_server).
+
+-export([start_link/0]).
+-export([init/1]).
+-export([handle_call/3]).
+-export([handle_cast/2]).
+-export([handle_info/2]).
+-export([code_change/3]).
+
+start_link() ->
+	gen_server:start_link(?MODULE, [], []).
+
+init([]) ->
+	case wait_ranch_server(5) of
+		{ok, Monitor} ->
+			{ok, Monitor};
+		{error, Reason} ->
+			{stop, Reason}
+	end.
+
+handle_call(_, _, Monitor) ->
+	{noreply, Monitor}.
+
+handle_cast(_, Monitor) ->
+	{noreply, Monitor}.
+
+handle_info({'DOWN', Monitor, process, _, Reason}, Monitor) ->
+	{stop, Reason, Monitor};
+handle_info(_, Monitor) ->
+	{noreply, Monitor}.
+
+code_change(_, Monitor, _) ->
+	{ok, Monitor}.
+
+-spec wait_ranch_server(non_neg_integer()) -> {ok, reference()} | {error, term()}.
+wait_ranch_server(N) ->
+	Monitor = monitor(process, ranch_server),
+	receive
+		{'DOWN', Monitor, process, _, noproc} when N > 0 ->
+			receive after 1000 -> ok end,
+			wait_ranch_server(N - 1);
+		{'DOWN', Monitor, process, _, Reason} ->
+			{error, Reason}
+	after 0 ->
```

Comment:
Is `after 0` correct? Isn't the `monitor` call asynchronous? Does it deliver the DOWN message immediately in the case of `noproc`?

Reply:
You may be right there, I might have mixed something up... Will check tomorrow.

Reply:
Hm, you're right. `link` is instantaneous when on the same node, `monitor` is async... Then how to do this in the best way?

Reply:
I think it should not be using a monitor, but just try to do `whereis(ranch_server)` until it gets a pid, and only THEN monitor it.

Reply:
You could sleep for say, 10ms between tries, to avoid busy looping.